### PR TITLE
NOISSUE - VerneMQ init script fixes

### PR DESCRIPTION
**Pull request by isavcic**
Signed-off-by: Ivan Savcic <isavcic@gmail.com>

### What does this do?

This fix to the VerneMQ Docker image init script properly handles the leaving of VerneMQ nodes from the cluster; once a node leaves it should not persist its state, because it can't join the cluster again as a new node (ie. cluster will keep no knowledge of it, because it left) with old state.

### Which issue(s) does this PR fix/relate to?

NOISSUE

### List any changes that modify/break current functionality

n/a

### Have you included tests for your changes?

No.

### Did you document any new/modified functionality?

It's self-documenting.

### Notes

None.

### CODE REVIEW

File: docker/vernemq/bin/vernemq.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ fi
 if env | grep "DOCKER_VERNEMQ_DISCOVERY_KUBERNETES" -q; then
     DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME=${DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME:-cluster.local}
     # Let's get the namespace if it isn't set
-    DOCKER_VERNEMQ_KUBERNETES_NAMESPACE=${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE:-`cat /var/run/secrets/kubernetes.io/serviceaccount/namespace`}
+    DOCKER_VERNEMQ_KUBERNETES_NAMESPACE=${DOCKER_VERNEMQ_KUBERNETES_NAMESPACE:-$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)}
```
Comment (drasko) [inline docker/vernemq/bin/vernemq.sh:52]: Is there a particular need for this change?

File: docker/vernemq/bin/vernemq.sh [hunk 2]
```diff
@@ -84,13 +86,13 @@ if [ -f /vernemq/etc/vernemq.conf.local ]; then
 else
     sed -i '/########## Start ##########/,/########## End ##########/d' /vernemq/etc/vernemq.conf
 
-    echo "########## Start ##########" >> /vernemq/etc/vernemq.conf
+    echo "########## Start ##########" >>/vernemq/etc/vernemq.conf
```
Comment (drasko) [inline docker/vernemq/bin/vernemq.sh:89]: Do not remove space after `>>`

File: docker/vernemq/bin/vernemq.sh [hunk 3]
```diff
@@ -84,13 +86,13 @@ if [ -f /vernemq/etc/vernemq.conf.local ]; then
...
 
-    env | grep DOCKER_VERNEMQ | grep -v 'DISCOVERY_NODE\|KUBERNETES\|SWARM\|COMPOSE\|DOCKER_VERNEMQ_USER' | cut -c 16- | awk '{match($0,/^[A-Z0-9_]*/)}{print tolower(substr($0,RSTART,RLENGTH)) substr($0,RLENGTH+1)}' | sed 's/__/./g' >> /vernemq/etc/vernemq.conf
+    env | grep DOCKER_VERNEMQ | grep -v 'DISCOVERY_NODE\|KUBERNETES\|SWARM\|COMPOSE\|DOCKER_VERNEMQ_USER' | cut -c 16- | awk '{match($0,/^[A-Z0-9_]*/)}{print tolower(substr($0,RSTART,RLENGTH)) substr($0,RLENGTH+1)}' | sed 's/__/./g' >>/vernemq/etc/vernemq.conf
```
Comment (drasko) [inline docker/vernemq/bin/vernemq.sh:91]: Same here

File: docker/vernemq/bin/vernemq.sh [hunk 4]
```diff
@@ -84,13 +86,13 @@ if [ -f /vernemq/etc/vernemq.conf.local ]; then
...
 
     users_are_set=$(env | grep DOCKER_VERNEMQ_USER)
     if [ ! -z "$users_are_set" ]; then
-        echo "vmq_passwd.password_file = /vernemq/etc/vmq.passwd" >> /vernemq/etc/vernemq.conf
+        echo "vmq_passwd.password_file = /vernemq/etc/vmq.passwd" >>/vernemq/etc/vernemq.conf
```
Comment (drasko) [inline docker/vernemq/bin/vernemq.sh:95]: Same here

**Review by drasko — CHANGES_REQUESTED** [review]

File: docker/vernemq/bin/vernemq.sh (hunk 2 above)
Comment (isavcic) [reply docker/vernemq/bin/vernemq.sh:89]: This is contrary to what shfmt, the de facto shell script formatter does.

File: docker/vernemq/bin/vernemq.sh (hunk 1 above)
Comment (isavcic) [reply docker/vernemq/bin/vernemq.sh:52]: This was an upstream change, I believe.

**Comment by drasko** [comment]
@isavcic I have checked here: https://github.com/mvdan/sh/issues/243 and here: https://github.com/mvdan/sh/pull/246. After reding this,  I feel that your first formatting was actually better and I would lean towards it, especially because several popular formatters use it as a default.

Sorry for the additional work, but please revert to previous formatting.

Otherwise PR looks quite good.

**Review by dborovcanin — DISMISSED** [review]

**Review by blokovi — DISMISSED** [review]

**Review by drasko — DISMISSED** [review]
LGTM

**Review by blokovi — DISMISSED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**